### ISSUE TITLE
func_proxies: dict=None, Type "None" cannot be assigned to type "dict[Unknown, Unknown]"

### ISSUE THREAD

**Issue by Baitwo02** [microsoft/pyright#3777]
```python
def foo(func_proxies: dict=None):
    pass
```

```
[Pyright reportGeneralTypeIssues] [E] Expression of type "None" cannot be assigned to parameter of type "dict[Unknown, Unknown]"
  Type "None" cannot be assigned to type "dict[Unknown, Unknown]"
```

what cai i do?
It haves no problem when i'm writing this in pycharm

**Comment by erictraut** [comment]
You have indicated here that the parameter allows only values of type `dict`, but you are attempting to assign it a value of `None` which is incompatible. If your intent is for the parameter to accept either a `dict` or `None`, you should indicate that using a union.

```python
def foo(func_proxies: dict | None = None):
    pass
```

Earlier in the development of the Python static type system, it was common to use `= None` as a shortcut to indicate that the type also supports `None`, but this practice was deemed inadvisable, and PEP 484 was amended a couple years ago to indicate that type checkers should no longer accept this. It sounds like PyCharm has not been updated to conform to the new guidance.

Pyright offers an option to switch back to the old (deprecated) behavior. You can create a `pyrightconfig.json` file and add the following:
```json
{
    "strictParameterNoneValue": false
}
```

I recommend switching away from this deprecated practice.

**Comment by erictraut** [comment]
If you have general questions about pyright or static type checking in Python, the discussion tab is a more appropriate place to post. Of course, if you suspect that you have found a real bug, you're welcome to file a bug report in the issue tracker.
